dali/base/dautils.cpp

```diff
@@ -456,28 +456,62 @@ void CDfsLogicalFileName::normalizeName(const char *name, StringAttr &res, bool
     res.set(str);
 }
 
-bool isExternalFile(const char *name)
+bool isExternalFile(const char *name, StringAttr &lfn, bool strict, unsigned &tailpos)
 {
     // TODO Should check the name is a valid OS filename
     if ('~' == *name) // allowed 1 leading ~
+    {
         name++;
+        if (!strict)
+            skipSp(name);
+    }
 
     bool retVal = memicmp(name,EXTERNAL_SCOPE "::",sizeof(EXTERNAL_SCOPE "::")-1)==0;
-    const char *s=strstr(name,"::");
-    if (retVal && s)
+    if (retVal)
     {
-        const char *s1 = s+2;
-        const char *ns1 = strstr(s1,"::");
-        if (!ns1)
-            retVal = false;
-        else
+        lfn.clear();
+        StringBuffer str;
+        const char *s=strstr(name,"::");
+        if (s)
         {
-            StringBuffer nodename;
-            nodename.append(ns1-s1,s1);
-            SocketEndpoint ep(nodename.str());
-            if (ep.isNull())
+            normalizeScope(name, name, s-name, str, strict);
+            const char *s1 = s+2;
+            const char *ns1 = strstr(s1,"::");
+            if (!ns1)
                 retVal = false;
+            else
+            {
+                if (!strict)
+                    skipSp(s1);
+
+                StringBuffer nodename;
+                nodename.append(ns1-s1,s1);
+                if (!strict)
+                    nodename.clip();
+                SocketEndpoint ep(nodename.str());
+                if (ep.isNull())
```

Comment:
There's quite a lot of duplicated code now above (common with normalizeName).
It would be good to common them up, which will be easier once parseExternal() is a member.